app/src/main/java/ch/epfl/qedit/view/home/HomePopUp.java

```diff
@@ -0,0 +1,157 @@
+    private Context context;
+    private User user;
+    private RecyclerView.Adapter customAdapter;
+
+    public HomePopUp(Context context, User user, RecyclerView.Adapter customAdapter) {
+        this.context = context;
+        this.user = user;
+        this.customAdapter = customAdapter;
+    }
+
+    public AlertDialog popUpEdit(final String oldValue, final int position) {
+        final AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        builder.setTitle("Give a new name... Or the same one");
+
+        final EditText input = new EditText(context);
+        input.setInputType(InputType.TYPE_CLASS_TEXT);
+        builder.setView(input);
+
+        setNegativeButton(builder);
+
+        builder.setPositiveButton(
+                "Done",
+                new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        user.updateQuizOnValue(oldValue, input.getText().toString());
+                        customAdapter.notifyItemChanged(position);
+                    }
+                });
+
+        return errorDialog(builder, input);
+    }
+
+    private void setNegativeButton(AlertDialog.Builder builder) {
+        builder.setNegativeButton(
+                "Cancel",
+                new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        dialog.cancel();
+                    }
+                });
+    }
+
+    public AlertDialog errorDialog(AlertDialog.Builder builder, final EditText editText) {
+        final AlertDialog alertDialog = builder.create();
+        alertDialog.create();
+        alertDialog.getButton(AlertDialog.BUTTON_POSITIVE).setClickable(false);
+        editText.setError(errorBlank);
+        alertDialog.getButton(AlertDialog.BUTTON_POSITIVE).setTextColor(Color.WHITE);
+        editText.addTextChangedListener(new CustomTextWatcher(editText, alertDialog));
+
+        return alertDialog;
+    }
+
+    public AlertDialog popUpWarningDelete(final String title, final int position) {
+        final AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        builder.setTitle(
+                "Are you sure you want to delete, deleting will delete all questions from this quiz");
+
+        setNegativeButton(builder);
+
+        builder.setPositiveButton(
+                "Yes",
+                new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        user.removeQuizOnValue(title);
+                        customAdapter.notifyItemRemoved(position);
+                    }
+                });
+
+        final AlertDialog alertDialog = builder.create();
+        alertDialog.create();
+        return alertDialog;
+    }
+
+    public void addPopUp() {
```

Comment:
I feel like the interface of that class is not very consistent. Sometimes, the methods return an `AlertDialog` (for example `popUpEdit`), and sometimes they don't (like this one). Furthermore, if we want clear interfaces, we should not expose unnecessary methods. I am thinking about `errorDialog`, which is only used internally, but public.

Reply:
I was for not exposing the entire class but code climate didn't like it...

Reply:
Really ? I don't see why it might be complaining...